File: src/models/lists/Shops.java
package models.lists;

import java.io.Serializable;
import java.util.List;

import models.Shop;

public class Shops extends Models<Shop> implements Serializable
{

    private static final int shopListSize = 40;

    public Shops()
    {
	super();
    }

    public Shops(Shops addFrom)
    {
	super(addFrom);
    }

    @Override
    public void initializeModel(List<Shop> models)
    {
	for (int i = 0; i < shopListSize; i++)
	{
	    Shop s = new Shop(i);
	    s.getValuesFromROM();
	    models.add(s);
	}
    }

    @Override
    public void saveModels()
    {
	for (Shop s : models)
	    s.writeValuesToROM();
    }

}

File: src/application/models/lists/Shops.kt
package application.models.lists

import application.models.Shop
import java.io.Serializable

class Shops : Models<Shop>, Serializable
{

	constructor() : super()
	constructor(addFrom: Shops) : super(addFrom)

	override fun initialize(models: ModelsList<Shop>)
	{
		for (i in 0..shopListSize - 1)
		{
			val s = Shop(i)
			s.getValuesFromROM()
			models.add(s)
		}
	}

	override fun saveModels() = models.forEach { it.writeValuesToROM() }

	companion object
	{
		private val shopListSize = 40
	}

}
